src/she/sdl2/sdl2_display.cpp

```diff
@@ -0,0 +1,363 @@
+    she::sdl::screen = newSurface;
+  }
+
+  Surface* SDL2Display::getSurface()
+  {
+    return m_surface;
+  }
+
+  void SDL2Display::present() {
+    if (!m_dirty)
+      return;
+    m_dirty = false;
+    SDL_UpdateWindowSurface(m_window);
+  }
+
+  void SDL2Display::flip(const gfx::Rect& bounds)
+  {
+    m_dirty = true;
+
+    auto nativeSurface = SDL_GetWindowSurface(m_window);
+    SDL_Rect rect {bounds.x, bounds.y, bounds.w, bounds.h};
+    SDL_Rect dst  {bounds.x * m_scale, bounds.y * m_scale, bounds.w * m_scale, bounds.h * m_scale};
+    SDL_BlitScaled((SDL_Surface*)m_surface->nativeHandle(), &rect, nativeSurface, &dst);
+  }
+
+  void SDL2Display::maximize()
+  {
+    SDL_MinimizeWindow(m_window);
+  }
+
+  bool SDL2Display::isMaximized() const
+  {
+    return sdl::isMaximized;
+  }
+
+  bool SDL2Display::isMinimized() const
+  {
+    return sdl::isMinimized;
+  }
+
+  void SDL2Display::setTitleBar(const std::string& title)
+  {
+    SDL_SetWindowTitle(m_window, title.c_str());
+  }
+
+  NativeCursor SDL2Display::nativeMouseCursor()
+  {
+    return m_nativeCursor;
+  }
+
+  bool SDL2Display::setNativeMouseCursor(NativeCursor cursor)
+  {
+    return false;
+
+    if (cursor == m_nativeCursor) {
+      return true;
+    }
+
+    if (m_cursor)
+      SDL_FreeCursor(m_cursor);
+    m_cursor = SDL_DISABLE;
+
+    switch (cursor) {
+    case kNoCursor: break;
+    case kArrowCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_ARROW); break;
+    case kIBeamCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_IBEAM); break;
+    case kWaitCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_WAIT); break;
+    // case kHelpCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HELP); break;
+    case kForbiddenCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_NO); break;
+    case kMoveCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_HAND); break;
+    // case kLinkCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_LINK); break;
+    case kSizeNSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENS); break;
+    case kSizeWECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEWE); break;
+    // case kSizeNCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEN); break;
+    // case kSizeNECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENE); break;
+    // case kSizeECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEE); break;
+    // case kSizeSECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESE); break;
+    // case kSizeSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZES); break;
+    // case kSizeSWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESW); break;
+    // case kSizeWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEW); break;
+    // case kSizeNWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENW); break;
```

Comment:
It seems that many of these cursors are unavailable with SDL. The commented code can probably be removed.